inference-engine/src/mkldnn_plugin/nodes/mkldnn_topk_node.cpp

```diff
@@ -2,33 +2,1287 @@
+
+        load_scalar(xmm_val_l, ptr[reg_dst + offset_l * jcp_.data_size], jcp_.data_type);
+        load_scalar(xmm_idx_l, ptr[reg_dst_idx + offset_l * sizeof(int)], memory::data_type::s32);
+        if (r != jcp_.top_k) {
+            load_scalar(xmm_val_r, ptr[reg_dst + offset_r * jcp_.data_size], jcp_.data_type);
+            load_scalar(xmm_idx_r, ptr[reg_dst_idx + offset_r * sizeof(int)], memory::data_type::s32);
+        }
+
+        swap_scalar(cmp_val);
+
+        store_scalar(ptr[reg_dst + offset_l * jcp_.data_size], xmm_val_l, jcp_.data_type);
+        store_scalar(ptr[reg_dst_idx + offset_l * sizeof(int)], xmm_idx_l, memory::data_type::s32);
+        if (r != jcp_.top_k) {
+            store_scalar(ptr[reg_dst + offset_r * jcp_.data_size], xmm_val_r, jcp_.data_type);
+            store_scalar(ptr[reg_dst_idx + offset_r * sizeof(int)], xmm_idx_r, memory::data_type::s32);
+        }
+    }
+
+    inline void swap_scalar(bool cmp_val = true) {
+        if (isa == avx512_common) {
+            if (cmp_val)
+                vcmpps(k_mask, xmm_val_l, xmm_val_r, cmp_flg);
+            else
+                vcmpps(k_mask, xmm_idx_l, xmm_idx_r, _cmp_le_os);
+
+            uni_vmovups(xmm_tmp, xmm_val_l);
+            vblendmps(xmm_val_l | k_mask, xmm_val_r, xmm_val_l);
+            vblendmps(xmm_val_r | k_mask, xmm_tmp, xmm_val_r);
+
+            uni_vmovups(xmm_tmp, xmm_idx_l);
+            vblendmps(xmm_idx_l | k_mask, xmm_idx_r, xmm_idx_l);
+            vblendmps(xmm_idx_r | k_mask, xmm_tmp, xmm_idx_r);
+        } else {
+            if (cmp_val) {
+                if (cmp_flg == _cmp_nlt_us)
+                    vcmpge_oqps(xmm_mask, xmm_val_l, xmm_val_r);
+                else
+                    vcmple_oqps(xmm_mask, xmm_val_l, xmm_val_r);
+            } else {
+                vcmple_oqps(xmm_mask, xmm_idx_l, xmm_idx_r);
+            }
+
+            uni_vmovups(xmm_tmp, xmm_val_l);
+            vblendvps(xmm_val_l, xmm_val_r, xmm_val_l, xmm_mask);
+            vblendvps(xmm_val_r, xmm_tmp, xmm_val_r, xmm_mask);
+
+            uni_vmovups(xmm_tmp, xmm_idx_l);
+            vblendvps(xmm_idx_l, xmm_idx_r, xmm_idx_l, xmm_mask);
+            vblendvps(xmm_idx_r, xmm_tmp, xmm_idx_r, xmm_mask);
+        }
+    }
+
+    inline void swap_scalar(int l, int r, bool cmp_val = true) {
+        if (isa == avx512_common) {
+            if (cmp_val)
+                vcmpps(k_mask, xmm_val(l), xmm_val(r), cmp_flg);
+            else
+                vcmpps(k_mask, xmm_idx(l), xmm_idx(r), _cmp_le_os);
+
+            uni_vmovups(xmm_tmp, xmm_val(l));
+            vblendmps(xmm_val(l) | k_mask, xmm_val(r), xmm_val(l));
+            vblendmps(xmm_val(r) | k_mask, xmm_tmp, xmm_val(r));
+
+            uni_vmovups(xmm_tmp, xmm_idx(l));
+            vblendmps(xmm_idx(l) | k_mask, xmm_idx(r), xmm_idx(l));
+            vblendmps(xmm_idx(r) | k_mask, xmm_tmp, xmm_idx(r));
+        } else {
```

Comment:
As I can see `else` branch is intended to handle `sse` isa as well. Am I right? If so seems this won't because you are using instructions with vex prefix.

Reply:
Fixed. I've checked the sse instructions, and added the sse branch. Thanks Dmitry!